### PR TITLE
Misuv 3702 mfa charge summary

### PR DESCRIPTION
**Pull request by joesheridan**

### CODE REVIEW

File: it/controllers/ChargeSummaryControllerISpec.scala [hunk 1]
```diff
@@ -412,6 +412,240 @@ class ChargeSummaryControllerISpec extends ComponentSpecBase {
     }
   }
 
+  "MFADebits feature on Charge Summary Page" should {
+    val financialDetailsUnpaidMFA = Json.obj(
```
Comment (edward-dennis-ey) [inline it/controllers/ChargeSummaryControllerISpec.scala:416]: These variables could potentially go in the FinancialDetailsIntegrationTestConstants

File: it/controllers/ChargeSummaryControllerISpec.scala [hunk 2]
```diff
@@ -412,6 +412,240 @@ class ChargeSummaryControllerISpec extends ComponentSpecBase {
...
+      enable(PaymentAllocation)
+
+      Given("I wiremock stub a successful Income Source Details response with property only")
+      IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesAndPropertyResponse)
+
+      And("I wiremock stub a single financial transaction response")
+      IncomeTaxViewChangeStub.stubGetFinancialDetailsByDateRange(testNino)(OK, financialDetailsUnpaidMFA)
+
+      And("I wiremock stub a charge history response")
+      IncomeTaxViewChangeStub.stubChargeHistoryResponse(testMtditid, "1040000124")(OK, testChargeHistoryJson(testMtditid, "1040000124", 2500))
```
Comment (edward-dennis-ey) [inline it/controllers/ChargeSummaryControllerISpec.scala:539]: You could reduce duplication by using a setup function for these 2 tests

File: it/controllers/ChargeSummaryControllerISpec.scala [hunk 3]
```diff
@@ -412,6 +412,240 @@ class ChargeSummaryControllerISpec extends ComponentSpecBase {
...
+          Some("12345-credId"), Some("Individual"), None
+        )(FakeRequest()),
+        DocumentDetailWithDueDate(
+          documentDetail = DocumentDetail(
+            taxYear = "2018",
+            transactionId = "1040000124",
+            documentDescription = Some("TRM New Charge"),
+            documentText = Some("documentText"),
+            originalAmount = Some(1200),
+            outstandingAmount = Some(0),
```
Comment (edward-dennis-ey) [inline it/controllers/ChargeSummaryControllerISpec.scala:634]: This figure is the only one that changes? Could pass it in instead?

Comment (joesheridan) [reply it/controllers/ChargeSummaryControllerISpec.scala:634]: sure will do

**Review by edward-dennis-ey — APPROVED** [review]

**Pull request merged**